index.js

```diff
@@ -6,31 +6,56 @@ const getCodePoint = character => `\\u{${character.codePointAt(0).toString(16)}}
 export class JSONError extends Error {
 	name = 'JSONError';
 	fileName;
-	codeFrame;
-	rawCodeFrame;
+	#input;
+	#jsonParseError;
 	#message;
+	#codeFrame;
+	#rawCodeFrame;
 
-	constructor(message) {
+	constructor({jsonParseError, fileName, input}) {
```

Comment:
The error is exported, so this is technically a breaking change. I doubt anyone is using it directly though. But just in case, maybe throw an error if it's used with a non-object argument?

Reply:
How about https://github.com/sindresorhus/parse-json/pull/50/commits/04c3224d144e69367ad0d67e1c6403dda7cd6b4e

Reply:
`codeFrame` and `rawCodeFrame` are marked `readonly` in types, but they are assignable, should I add setters?

Reply:
> codeFrame and rawCodeFrame are marked readonly in types, but they are technically
assignable, should I add setters?

Nah. Too much of an edge-case.